pkg/s3io/s3io.go

```diff
@@ -30,7 +30,7 @@ func IsS3Path(path string) bool {
 }
 
 func parsePath(path string) (bucket, key string, err error) {
-	u, err := iosrc.ParseURI(path)
+	u, err := iosrc.Parse(path)
```

Comment:
Can you leave URI in pkg/iosrc if you use net/url.Parse here?

Reply:
It could... I don't think its bad uri is in its own package (kind of makes sense imho).

Reply:
How was this not cause compilation errors?!

Reply:
Not opposed going your way though.

Reply:
After having the weekend away from this I'm going to get rid of the move URI to its own package. If I want to do this, this should go in another pr.